### ISSUE TITLE
Gradle build fails on Flutter with Kotlin support

### ISSUE THREAD

**Issue by brazc** [heremaps/here-sdk-examples#78]
After migrating to Flutter 2, it seems that my app fails to compile.

```
e: C:\development\flutter_app\plugins\here_sdk\android\src\main\kotlin\com\here\mapview\MapController.kt: (31, 1): Class 'MapController' is not abstract and does not implement abstract member public abstract fun onActivityPaused(p0: Activity): Unit defined in android.app.Application.ActivityLifecycleCallbacks
e: C:\development\flutter_app\plugins\here_sdk\android\src\main\kotlin\com\here\mapview\MapController.kt: (120, 5): 'onActivityPaused' overrides nothing
e:C:\development\flutter_app\plugins\here_sdk\android\src\main\kotlin\com\here\mapview\MapController.kt: (124, 5): 'onActivityResumed' overrides nothing
e: C:\development\flutter_app\plugins\here_sdk\android\src\main\kotlin\com\here\mapview\MapController.kt: (128, 5): 'onActivityStarted' overrides nothing
e: C:\development\flutter_app\plugins\here_sdk\android\src\main\kotlin\com\here\mapview\MapController.kt: (135, 5): 'onActivityDestroyed' overrides nothing
e: C:\development\flutter_app\plugins\here_sdk\android\src\main\kotlin\com\here\mapview\MapController.kt: (139, 5): 'onActivitySaveInstanceState' overrides nothing
e: C:\development\flutter_app\plugins\here_sdk\android\src\main\kotlin\com\here\mapview\MapController.kt: (143, 5): 'onActivityStopped' overrides nothing
e: C:\development\flutter_app\plugins\here_sdk\android\src\main\kotlin\com\here\mapview\MapController.kt: (150, 5): 'onActivityCreated' overrides nothing

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':here_sdk:compileDebugKotlin'.
> Compilation error. See log for more details
```

Flutter version 2.0.3
Dart 2.12.2
Gradle 4.1.3
Kotlin 1.4.32

**Comment by brazc** [comment]
@datasun  does this issue occur for you? It's critical and I'm not sure what exactly caused the issue. I've tried numerous Kotlin versions and all result in this.

The example apps do not have this issue.

**Comment by datasun** [comment]
@brazc From Android Studio IDE you can convert the example apps to Kotlin. Is this then also working for you?

**Comment by brazc** [comment]
@datasun making a new isolated app doesn't compile as the dart compiler complains about here_sdk `4.7.0` and ffi (0.1.3) not supporting sound null-safety.

```
Running Gradle task 'assembleDebug'...
Error: Cannot run with sound null safety, because the following dependencies
don't support null safety:

 - package:here_sdk
 - package:ffi

For solutions, see https://dart.dev/go/unsound-null-safety


FAILURE: Build failed with an exception.
```


```
brazc@conti ~ % flutter doctor -v
[✓] Flutter (Channel stable, 2.0.5, on macOS 11.2.3 20D91 darwin-x64, locale en-US)
    • Flutter version 2.0.5 at /Users/brazc/dev/flutter
    • Framework revision adc687823a (5 days ago), 2021-04-16 09:40:20 -0700
    • Engine revision b09f014e96
    • Dart version 2.12.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/brazc/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
```

Downgrading flutter version from `2.0.5` to `2.0.2` is tedious (I should probably install [fvm](https://github.com/leoafarias/fvm) at this point) but if all else fails I'll try with that.
I'm not exactly sure when you mean the latest SDK release, the one that's available to download is `4.7.0.0`

**Comment by datasun** [comment]
Thanks. The fix I mentioned above was already included in 4.7.0., but seeing now that the compiler complains about missing null-safety ... - that sounds like an issue with FFI. We do not support FFI 1.0 yet. We also do not support null-safety yet. If your project contains other 3rd party plugins that depend on FFI 1.0 then we cannot do anything until the HERE SDK also supports FFI 1.0 (which is planned for one of the next releases). The problem is that other customers still use FFI 0.1.3 - so it's kind of a dependency hell. 

If this is not the issue here, we could still take a look into your isolated app. It doesn't matter if it compiles or not ... seeing the project files and repeating the steps should be enough.

However, I would like to add that we plan to support null safety with HERE SDK v4.8.0

**Comment by brazc** [comment]
I guess it's dependency hell for me, but I assume this can be solved by downgrading to 2.0.2, given its stated that the SDK is compatible with Flutter 2.0.2. A lot of the dependencies I use have already moved on to FFI 1.0 and support null-safety. So this means downgrading all my other dependencies as well.

Not a big issue as long as the app compiles.

**Comment by RohanSAbraham** [comment]
My team is facing the same issue

**Comment by datasun** [comment]
This should be fixed with HERE SDK 4.8 release that is planned to happen in around 4 weeks.
